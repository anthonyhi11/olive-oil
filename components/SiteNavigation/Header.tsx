import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import styles from "../../styles/header.module.css";

//TODO: link isn't styled correctly if I use
// LINK from next. but if i use Nav.Link, it has all the style perfect but
// it reloads the page.
// Probably just leave as is if planning on to use the library

interface IHeaderProps {
	active: string;
}

function Header({ active }: IHeaderProps) {
	return (
		<Navbar variant="light" sticky="top">
			<Navbar.Brand className={styles.headerNavBrand} href="/">
				Olive Oil
			</Navbar.Brand>

			<Nav activeKey={active}>
				<Nav.Item>
					<Nav.Link href="/">Home</Nav.Link>
				</Nav.Item>
				<Nav.Item>
					<Nav.Link href="/about">About</Nav.Link>
				</Nav.Item>
				<Nav.Item>
					<Nav.Link href="/scraper">Scraper</Nav.Link>
				</Nav.Item>
				<Nav.Item>
					<Nav.Link href="/contact">Contact</Nav.Link>
				</Nav.Item>
			</Nav>
		</Navbar>
	);
}

export default Header;
