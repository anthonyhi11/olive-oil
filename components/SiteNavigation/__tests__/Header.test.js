import React from "react";
import {render} from "@testing-library/react";
import Header from "../Header";

test("renders and matches snapshot", () => {
	const {container} = render(<Header active="/" />);
	expect(container).toMatchSnapshot();
});
