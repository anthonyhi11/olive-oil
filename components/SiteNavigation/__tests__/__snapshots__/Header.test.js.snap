// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-light sticky-top"
    gb="light"
  >
    <a
      class="navbar-brand"
      href="/"
    >
      Olive Oil
    </a>
    <div
      class="navbar-nav"
    >
      <div
        class="nav-item"
      >
        <a
          class="nav-link active"
          data-rb-event-key="/"
          href="/"
        >
          Home
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rb-event-key="/about"
          href="/about"
        >
          About
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rb-event-key="/scraper"
          href="/scraper"
        >
          Scraper
        </a>
      </div>
      <div
        class="nav-item"
      >
        <a
          class="nav-link"
          data-rb-event-key="/contact"
          href="/contact"
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</div>
`;
